lib/valueflow.cpp

```diff
@@ -2339,11 +2339,11 @@ static bool bifurcate(const Token* tok, const std::set<nonneg int>& varids, cons
 }
 
 struct ValueFlowAnalyzer : Analyzer {
-    const TokenList* tokenlist;
+    const TokenList& tokenlist;
```

Comment:
this is still a reference I thought we agree to avoid references for members.

Reply:
See https://github.com/danmar/cppcheck/pull/4868#discussion_r1131282725 and https://github.com/danmar/cppcheck/pull/4868#issuecomment-1462367067 - no definite decision yet.

Again I wonder if we could keep this temporarily (we already use it across the code) and I can finish the remaining changes related to this. Shortly after I do away with the pointers in a batch with properly having tested the potential several approaches.